src/composables/transactionList.ts

```diff
@@ -0,0 +1,337 @@
+import { orderBy, uniqBy } from 'lodash-es';
+import {
+  computed, onMounted,
+  ref,
+  watch,
+} from 'vue';
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+
+import type {
+  ITransaction,
+  ITransactionsState,
+  IAccountTransactionsState,
+  IDefaultComposableOptions,
+} from '../types';
+import { useMiddleware } from './middleware';
+import {
+  AEX9_TRANSFER_EVENT,
+  TX_DIRECTION,
+  fetchJson,
+  getLocalStorageItem,
+  setLocalStorageItem,
+} from '../popup/utils';
+import { useAccounts } from './accounts';
+import JsonBig from '../lib/json-big';
+import { useAeSdk } from './aeSdk';
+
+const LOCAL_STORAGE_KEY = 'transactions';
+
+const isInitialized = ref(false);
+
+const transactions = ref<IAccountTransactionsState>({});
+
+function generateEmptyTransactionState(): ITransactionsState {
+  return {
+    loaded: [],
+    nextPageUrl: '',
+    pending: {},
+    tipWithdrawnTransactions: [],
+  };
+}
+
+function getAccountTransactionsState(address: Encoded.AccountAddress): ITransactionsState {
+  return transactions.value[address] || generateEmptyTransactionState();
+}
+
+function createTransactionsState(address: Encoded.AccountAddress) {
```

Comment:
As this function has an `if` inside I think this can be renamed to `ensureAccountTransactionStateExists`.